docs/patterns/ci-cd-10000-feet.md

```diff
@@ -0,0 +1,170 @@
+See also [Standard's Just Integration][std-just] for practical guidance.
+
+We can think of these task runners as the _views_ within the [MVC pattern][mvc] into the _actual_ task definitions.
+
+The task definitions are typically located in a "Tasks" (`tasks.nix`) Cell Block of the [`tasks` Block Type (`blockType.tasks`)][blocktype-tasks].
+
+Ensuring an expected shape of the code base is the annoying, but relatively _cheap_ paperwork (through repoautomation), that we do in order to unlock the next phase: the building phase.
+
+In the case that we need to ensure sequenciality, the register can be filled in sequential chuncks of targets that can be run in parallel.
+
+### Building Phase
+
+```yaml
+Register: __std.ci.build # list (order) of attrs (parallel) of derivations
+```
+
+Since [Nix's superpowers][nix-superpower] are derivations, we rely on this fundamental capability to built up a fully declarative, parallelized and reproducible build pipeline.
+
+This pipeline should be typically backed by a powerful build server farm.
+
+So all that we really do here is to fill the CI register with build targets.
+
+However, it is important that Unit Tests are embedded as part of the Nix build pipeline.
+This ensures for everyones sanity, that only a unit-tested build-arifact counts as a successful build.
+
+In the case that we need to ensure sequenciality, the register can be filled in sequential chuncks of targets that can be built in parallel.
+
+### Deployment Phase
+
+```yaml
+Register: __std.ci.deploy # list (order) of attrs (parallel) of deployment actions
+```
+
+The deployment phase schedules deployment artifacts via schematic target state (data-)manifests.
+It pushes the target state onto a (potentially stateful) control loop via a (potentially remote) API interface.
+
+The specific Block Type Action pushed to the `__std.ci.deploy` Register determins _how_ this process is to be accomplished.
+
+> **Push vs Pull Workflows**
+>
+> This concept enacts a push-based workflow as opposed to a pull-based workflow.
+> Push-based workflows are predominant in DevOps, but they actually puncture security holes into your target infrastructure.
+> There is no way around _listening_ on some port for the push to happen.
+>
+> Pull-based workflows don't suffer from this structural shortcoming and we may explore them closer in the future.
+
+> **Orchestration vs Choreography**
+>
+> This concept enacts a form of orchestration.
+> For increasingly complex infrastructure, there is little room for fogoing the benefits of orchestration over choreography, such as having a tigthened control loop.
+>
+> Typically, push-based workflows use orchestration, because the can afford state locally and, hence, cheaply.
+>
+> Typically, pull-based wowkflows use choreography, because they can't orchestrate without global (i.e. remote) state and, hence, complexity.
```

Comment:
```suggestion
> Typically, pull-based workflows use choreography, because they can't orchestrate without global (i.e. remote) state and, hence, complexity.
```